src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -256,19 +290,23 @@ export const CriteriaGroupView: React.FC<CriteriaGroupViewProps> = (props) => {
           ]
         ),
         div([
-          (criteriaGroup.criteria.length !== 0 &&
-            _.map(
-              createCriteriaViewComponent((criteria: AnyCriteria) =>
-                updateCohort(_.set(`criteriaGroups.${index}.criteria`, _.without([criteria], criteriaGroup.criteria)))
-              ),
-              criteriaGroup.criteria
-            )) ||
-            div({ style: { marginTop: 15 } }, [
-              div({ style: { fontWeight: 'bold', fontStyle: 'italic' } }, ['No criteria yet']),
-              div({ style: { fontStyle: 'italic', marginTop: narrowMargin } }, [
-                "You can add a criteria by clicking on 'Add criteria'",
+          criteriaGroup.criteria.length !== 0
+            ? _.map(
+                ([i, criteria]: [number, AnyCriteria]) =>
+                  h(CriteriaView, {
+                    deleteCriteria,
+                    updateCriteria: (criteria) => updateCriteria(criteria, i),
```

Comment:
I think that `criteria` here is the same as `criteria` on line 295. For delete, the code creates a single lambda (over `updateCohort`) that's passed to all criteria view objects. For `update`, the code is creating a separate lambda for each object, so it could capture `criteria` since it's in scope, and this method could be `() => void`.

Or if `updateCriteria` is changed to look up the index using `id`, then it could be shared as `delete` is.

If you do keep `updateCriteria` as-is, it would be better to use a different name here (other than `criteria` so it's clear that this name is different (from a scope perspective) than the one on line 295.

Reply:
Its not the same criteria, it is the changed version of the criteria. I can change the name.

Reply:
> Its not the same criteria, it is the changed version of the criteria.

So there's a case where this criteria doesn't match the one generated by the `toIndexPairs`? How can that happen? If `criteriaGroup.criteria` changes, doesn't the Select get re-rendered?